vme/src/dilexp.cpp

```diff
@@ -1773,26 +1767,20 @@ void dilfe_ldstr(dilprg *p)
                     {
                         szonelog(p->frame->tmpl->zone, "DIL '%s' attempted to loadstr n illegal file name.", p->frame->tmpl->prgname);
                         v->val.num = FALSE;
-                        ;
                     }
                     else
                     {
                         sstr = nullptr;
-                        if (p->frame[0].tmpl->zone->getDILFilePath())
-                        {
-                            strcpy(filename, p->frame[0].tmpl->zone->getDILFilePath());
-                        }
-                        else
-                        {
-                            strcpy(filename, g_cServerConfig.getDILFileDir().c_str());
-                        }
-                        strcat(filename, "/strings/");
-                        strcat(filename, (char *)v1->val.ptr);
-                        v->val.num = load_string(filename, &sstr);
+                        std::filesystem::path filename{p->frame[0].tmpl->zone->getDILFilePath().value_or(g_cServerConfig.getDILFileDir())};
+                        filename += "/strings/";
```

Comment:
Well if this doesn't work I'll see it easily in the log file.

Reply:
My thoughts exactly :-) :boom: